src/parquet/column/scanner.h

```diff
@@ -45,7 +45,7 @@ class Scanner {
       values_buffered_(0),
       reader_(reader) {
     // TODO: don't allocate for required fields
-    def_levels_.resize(reader->descr()->is_optional() ? batch_size_ : 0);
+    def_levels_.resize(!reader->descr()->is_required() ? batch_size_ : 0);
```

Comment:
Here  `is_optional()` (`max_definition_level_ > 0`) and `! is_required()` (`!(max_definition_level_ == 0)`) are equivalent statements (given precondition that `max_definition_level_` is never less than 0). I want to use `! is_required()` for clarity. Technically, definition values exist for both `optional` and `repeated` values. Just checking for `optional` here is less clear.